src/fork.rs

```diff
@@ -262,6 +262,46 @@ impl<S: ForkSource> ForkStorage<S> {
         // TODO: Update this file to use proper enumeration index value once it's exposed for forks via API
         Some(0_u64)
     }
+
+    pub fn dump_state(&self) -> SerializableForkStorage {
```

Comment:
Suggestion: maybe add doc-comments explaining the semantics of both functions to help future readers?